unofficial/c511027103.lua

```diff
@@ -0,0 +1,104 @@
+--ZW－雷神猛虎剣
+--ZW - Lightning Blade (anime)
+--Script by Rundas
+local s,id=GetID()
+function s.initial_effect(c)
+    --equip
+    local e1=Effect.CreateEffect(c)
+    e1:SetDescription(aux.Stringid(id,0))
+    e1:SetType(EFFECT_TYPE_IGNITION)
+    e1:SetProperty(EFFECT_FLAG_CARD_TARGET)
+    e1:SetCategory(CATEGORY_EQUIP)
+    e1:SetRange(LOCATION_HAND+LOCATION_MZONE)
+    e1:SetTarget(s.eqtg)
+    e1:SetOperation(s.eqop)
+    c:RegisterEffect(e1)
+    --protecc
+    local e2=Effect.CreateEffect(c)
+    e2:SetType(EFFECT_TYPE_SINGLE)
+    e2:SetProperty(EFFECT_FLAG_SINGLE_RANGE)
+    e2:SetCode(EFFECT_CANNOT_BE_EFFECT_TARGET)
+    e2:SetRange(LOCATION_SZONE)
+    e2:SetCondition(s.con)
+    e2:SetValue(s.val)
+    c:RegisterEffect(e2)
+    --destroy replace
+    local e3=Effect.CreateEffect(c)
+    e3:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_CONTINUOUS)
+    e3:SetCode(EFFECT_DESTROY_REPLACE)
+    e3:SetRange(LOCATION_SZONE)
+    e3:SetTarget(s.reptg)
+    e3:SetValue(s.repval)
+    e3:SetOperation(s.repop)
+    c:RegisterEffect(e3)
+end
+
+s.listed_names={56840427}
+
+--equip
+
+function s.eqtg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and chkc:IsFaceup() and chkc:IsCode(56840427) end
+	if chk==0 then return Duel.GetLocationCount(tp,LOCATION_SZONE)>0
+		and Duel.IsExistingTarget(aux.FilterFaceupFunction(Card.IsCode,56840427),tp,LOCATION_MZONE,0,1,nil) end
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_EQUIP)
+	Duel.SelectTarget(tp,aux.FilterFaceupFunction(Card.IsCode,56840427),tp,LOCATION_MZONE,0,1,1,nil)
+	Duel.SetOperationInfo(0,CATEGORY_EQUIP,e:GetHandler(),1,0,0)
+end
+
+function s.eqop(e,tp,eg,ep,ev,re,r,rp)
+	local c=e:GetHandler()
+	if not c:IsRelateToEffect(e) then return end
+	if c:IsLocation(LOCATION_MZONE) and c:IsFacedown() then return end
+	local tc=Duel.GetFirstTarget()
+	if Duel.GetLocationCount(tp,LOCATION_SZONE)<=0 or tc:GetControler()~=tp or tc:IsFacedown() or not tc:IsRelateToEffect(e) then
+		Duel.SendtoGrave(c,REASON_EFFECT)
+		return
+	end
+	Duel.Equip(tp,c,tc,true)
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_SINGLE)
+	e1:SetCode(EFFECT_EQUIP_LIMIT)
+	e1:SetReset(RESET_EVENT+RESETS_STANDARD)
+	e1:SetValue(s.eqlimit)
+	e1:SetLabelObject(tc)
+	c:RegisterEffect(e1)
+	--atkup
+	local e2=Effect.CreateEffect(c)
+	e2:SetType(EFFECT_TYPE_EQUIP)
+	e2:SetCode(EFFECT_UPDATE_ATTACK)
+	e2:SetValue(1200)
+	e2:SetReset(RESET_EVENT+RESETS_STANDARD)
+	c:RegisterEffect(e2)
+end
+
+function s.eqlimit(e,c)
+	return c==e:GetLabelObject()
+end
+
+--protecc
+
+function s.con(e,tp,eg,ep,ev,re,r,rp)
+	return e:GetHandler():GetEquipTarget()~=nil

```

Comment:
Due to the way _lua script_ works, you actually don't need ```~=nil``` here.
Basically, in lua, ```nil``` is ```false```, thus anything ```not nil``` is ```true```.